modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -695,10 +568,12 @@ const DomParser = (settings?: DomParserSettings, schema = Schema()): DomParser =
 
     // Fix invalid children or report invalid children in a contextual parsing
     if (validate && state.invalidChildren.length) {
-      if (!args.context) {
-        fixInvalidChildren(state);
+      if (args.context) {
```

Comment:
You mention `nothing new added, just gotta make sure the old tests still pass` but this is definitely something new as previously it never tried to fix an invalid context so we'll need some tests for this change.

Reply:
New test added to assert that this works, let me know if you want more tests just to be extra safe

Reply:
I'm okay with that, as the rest should largely be tested elsewhere.